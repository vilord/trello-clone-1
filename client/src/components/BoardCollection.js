import React from 'react';
import PropTypes from 'prop-types';
import BoardTile from './BoardTile';
import { Icon } from 'semantic-ui-react';

import './BoardCollection.css';

const BoardCollection = ({ icon, title, boards }) => (
  <div className="BoardCollection">
    <div>
      <Icon name={icon} size="large" />
      <h3>{title}</h3>
    </div>
    <div>
      {boards.map((board, i) => (
        <BoardTile key={`BoardTile-${i}`} {...board} />
      ))}
    </div>
  </div>
);

BoardCollection.propTypes = {
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  boards: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      favorite: PropTypes.bool.isRequired,
      theme: PropTypes.shape({
        color: PropTypes.string.isRequired,
        picture: PropTypes.string,
      }).isRequired,
    }),
  ).isRequired,
};

export default BoardCollection;
