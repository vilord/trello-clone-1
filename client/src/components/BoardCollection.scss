.BoardCollection {
  margin: 20px;
  width: 100%;
  max-width: 1300px;

  div {
    display: flex;
    flex-direction: row;
    align-items: center;

    h3 {
      display: inline;
      margin: 0;
      margin-left: 5px;
    }

    i {
      margin-left: 10px;
      font-weight: 900;
      color: #999;
    }
  }
}
