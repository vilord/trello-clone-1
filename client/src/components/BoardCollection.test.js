import React from 'react';
import { shallow } from 'enzyme';

import BoardCollection from './BoardCollection';

describe('BoardCollection Component', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  let component;
  let minProps;
  beforeEach(() => {
    minProps = {
      icon: 'star',
      title: 'Stared Boards',
      boards: [
        {
          id: 'y42LW46J9luq3Xq9XMly',
          title: 'Trello Clone',
          favorite: false,
          theme: {
            color: 'blue',
            picture: '',
          },
        },
      ],
    };
    component = shallow(<BoardCollection {...minProps} />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  it('sets the Icon using props.icon', () => {
    expect(component.find('Icon').prop('name')).toBe(minProps.icon);
  });

  it('sets the collection Title using props.title', () => {
    expect(component.find('h3').text()).toBe(minProps.title);
  });

  it('creates BoardTile elements using props.boards', () => {
    expect(component.find('BoardTile')).toHaveLength(1);
    component.setProps({ boards: [] })
    expect(component.find('BoardTile')).toHaveLength(0);
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
