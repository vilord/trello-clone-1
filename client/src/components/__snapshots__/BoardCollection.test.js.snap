// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardCollection Component matches snapshot 1`] = `
<div
  className="BoardCollection"
>
  <div>
    <Icon
      as="i"
      name="star"
      size="large"
    />
    <h3>
      Stared Boards
    </h3>
  </div>
  <div>
    <BoardTile
      favorite={false}
      id="y42LW46J9luq3Xq9XMly"
      key="BoardTile-0"
      theme={
        Object {
          "color": "blue",
          "picture": "",
        }
      }
      title="Trello Clone"
    />
  </div>
</div>
`;
